src/Analysis/GWASUIApp/CaseControlGWAS.jsx

```diff
@@ -511,15 +511,28 @@ const CaseControlGWAS = (props) => {
         }
     };
 
+    // This function takes in list of dichotomous covariates (cohort-pair based) and other regular covariates (concept-based)
+    // and produces the generic "variables" list that can be used as input argument in different cohort-middleware endpoints.
+    function getVariablesList(dichotomousCovariates, otherCovariates) {
+        const variables = [];
+        for (let dichotomousCovariate of dichotomousCovariates) {
+            const variableItem = {'variable_type': 'custom_dichotomous', 'cohort_ids': [dichotomousCovariate.cohort1, dichotomousCovariate.cohort2] };
+            variables.push(variableItem);
+        }
+        for (let otherCovariate of otherCovariates) {
+            const variableItem = {'variable_type': 'concept', 'concept_id': otherCovariate.concept_id };
+            variables.push(variableItem);
+        }
+        return {'variables': variables};
+    }
+
     async function fetchOverlapInfo() {
-        var conceptIds = [...selectedCovariates].map((val) => val.concept_id);
-        const conceptIdsPayload = { ConceptIds: conceptIds };
         const statsEndPoint = `${cohortMiddlewarePath}cohort-stats/check-overlap/by-source-id/${sourceId}/by-case-control-cohort-definition-ids/${caseCohortDefinitionId}/${controlCohortDefinitionId}/filter-by-concept-id-and-value/${hareConceptId}/${selectedHareValueAsConceptId}`;
         const reqBody = {
             method: 'POST',
             credentials: 'include',
             headers,
-            body: JSON.stringify(conceptIdsPayload),
+            body: JSON.stringify(getVariablesList(selectedDichotomousCovariates, selectedCovariates)),
```

Comment:
ah this is case-control situation ignore

Reply:
Even for the quantitative scenario, it would probably be useful to have the outcome phenotype as part of the variables list (especially in the argo-wrapper `requestBody` below). Then you can just take this list as is and use it in the CSV and attrition table endpoints. The downside is that you need to remember to skip it whenever you need only the covariates in your own code.

Reply:
@pieterlukasse i am now working under the assumption the outcome phenotype is included in the variable list for continuous and the `outcome` field will be the concept ID of the outcome continuous phenotype. What i need us to standardize is the name of the property to hold the provided name of the custom dichotomous variable so I can include it in the metadata yaml.

Reply:
we'll use `provided_name`